nautobot/dcim/models/devices.py

```diff
@@ -998,3 +1013,69 @@ def to_csv(self):
             self.domain,
             self.master.name if self.master else None,
         )
+
+
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "dynamic_groups",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "statuses",
+    "webhooks",
+)
+class RedundancyGroup(PrimaryModel, ConfigContextModel, StatusModel):
+    """
+    A RedundancyGroup represents a logical grouping of physical hardware for the purposes of indicating redundancy pairs of high-availability.
+    """
+
+    name = models.CharField(max_length=100, unique=True)
+
+    failover_strategy = models.CharField(
+        max_length=50, blank=True, choices=RedundancyGroupFailoverStrategyChoices, verbose_name="Failover strategy"
+    )
+
+    comments = models.TextField(blank=True)
+
+    secrets_group = models.ForeignKey(
+        to="extras.SecretsGroup",
+        on_delete=models.SET_NULL,
+        default=None,
+        blank=True,
+        null=True,
+    )
```

Comment:
I'm realizing we don't have a `SecretsGroupModelMixin` (akin to `StatusModel[Mixin])` to supply this by default. We might want to backlog a TODO for this so that this pattern doesn't need to proliferate by having to manually include this field on models and model forms.